### DISCUSSION TITLE
different arrow type by different values in adjacency matrix

### DISCUSSION THREAD

**Discussion by amm266** [networkx/networkx#5732] [Q&A] [answered]
hi.
I'm working on a simulation for fair allocation problem.
in this problem there are multiple types of envy: 
1. normal envy
2. relaxed envy (e.g. EFX)
 and so I want to have different arrow types for different values in the adjacency matrix to show envy type:
for example:
`adjacency_matrix = np.array([[0., 0., 0.],[2., 0., 0.], [1., 0., 0.]])`
arrows should be drawn diffrently (e.g. in different color) when value is 2 compare to when value is 1.
is there a way to do this?
i appreciate your help

**Comment by rossbar** [comment, marked as answer]
The simplest way to draw different subsets of edges with different properties is to use the `edgelist` to specify the subset of edges you want to draw. For example, if you wanted to draw edges with `envy==1` in blue and edges with `envy==2` in red, you could do something like (untested):

```python
for envy_val, color in zip((1, 2), ("blue", "red")):
    edges = [(u, v) for u, v, d in G.edges(data="envy") if d == envy_val]
    nx.draw_networkx_edges(G, pos, edgelist=edges, edge_color=color)
```

**Reply by amm266** [reply]
tanks a lot!
working code for anyone wondering:
```python
adjacency = np.array([[0., 0., 0.], [2., 0., 0.], [1., 0., 0.]])
g = nx.from_numpy_matrix(adjacency, create_using=nx.DiGraph)
pos = nx.spring_layout(g)
for envy_val, color in zip((1, 2), ("blue", "red")):
    edges = [(u, v) for u, v, d in g.edges.data(data='weight') if d == envy_val]
    nx.draw_networkx_edges(g, pos=pos, edgelist=edges, edge_color=color)
nx.draw_networkx_nodes(g, pos=pos)
nx.draw_networkx_labels(g, pos=pos)
plt.show()
```
